src/basic/terminal-util.c

```diff
@@ -643,7 +643,8 @@ int acquire_terminal(
         int r, wd = -1;
 
         assert(name);
-        assert(IN_SET(flags & ~ACQUIRE_TERMINAL_PERMISSIVE, ACQUIRE_TERMINAL_TRY, ACQUIRE_TERMINAL_FORCE, ACQUIRE_TERMINAL_WAIT));
+        assert(IN_SET(flags & _ACQUIRE_TERMINAL_MODE_MASK,
+                      ACQUIRE_TERMINAL_TRY, ACQUIRE_TERMINAL_FORCE, ACQUIRE_TERMINAL_WAIT));
```

Comment:
Maybe store this in a separate var then, to avoid filtering the _MASK multiple times below?

Reply:
And permit _WATCH_SIGTERM only if ACQUIRE_TERMINAL_WAIT?